### PR TITLE
feat: revoke user roles

### PR DESCRIPTION
**Pull request by maxwellgithinji**
Signed-off-by: maxwellgithinji <maxwellgithinji@gmail.com>

# Review Checklist

[Bugs](https://shields.io/)
[Features](https://shields.io/)
[Tests](https://shields.io/)

#### Summary*

- [ ] fix: Bug fix for `Add the items tackled here as checklists`
- [x] feat: Completed task
- [ ] chore: Incomplete task
  - [ ] test: Sub-task 1

#### Structure*

- [ ] The Pull Request has a `proper` title that conforms to our [MR title standards](https://gist.github.com/mikepea/863f63d6e37281e329f8)
- [ ] The Pull Request has one commit, and if there are more than one, they should be squashed
- [ ] The commit should have a proper title and a short description
- [ ] The commit must be signed off
- [ ] Unused imports are not present
- [ ] Dead code is not present
- [ ] Ensure dry running library to confirm changes

#### Tests

- [ ] Proper and high quality unit, integration and acceptance(if applicable) tests have been written
- [ ] The coverage threshold should not be lowered

### Sign off*

- [ ] All comments have been resolved by the reviewers
- [ ] Approved by Czar {replace_with_czar_name}
- [ ] Signed off by second reviewer {replace_with_name}
- [ ] Ensure all checks are done before merging :warning:
- [ ] All PRs needs to be signed before merging :warning:

#### N/B:

- Add a checklist if more than one item is done.
- Add screenshots and/or images where necessary
- Indicate any breakages caused in the UI :exclamation:
- Where necessary, indicate which issue the Pull Request solves (Closes #)
- Any new files are updated in the folder structure in the [README](../../README.md)

### CODE REVIEW

File: pkg/mycarehub/application/exceptions/custom_errors.go [hunk 1]
```diff
@@ -432,3 +432,12 @@ func GetUserPermissionsErr(err error) error {
 		Code:    int(GetUserPermissionsError),
 	}
 }
+
+// RevokeRolesErr returns an error message when the user roles' revoke fails
```
Comment (KathurimaKimathi) [inline pkg/mycarehub/application/exceptions/custom_errors.go:436]: Change to docstring to ``// RevokeRolesErr returns an error message when the user roles' revocation fails``

File: pkg/mycarehub/application/exceptions/error_messages.go [hunk 2]
```diff
@@ -143,4 +143,7 @@ const (
 	GetUserRolesErrorMsg = "unable to get user roles"
 	// GetUserPermissionsErrorMsg is the error message displayed when a user permission is not found
 	GetUserPermissionsErrorMsg = "unable to get user permissions"
+
+	// RevokeRolesErrorMsg is the error message displayed when a user roles' revoke fails
```
Comment (KathurimaKimathi) [inline pkg/mycarehub/application/exceptions/error_messages.go:147]: change ``revoke`` to ``revocation``

File: pkg/mycarehub/infrastructure/database/postgres/gorm/update.go [hunk 3]
```diff
@@ -704,3 +705,56 @@ func (db *PGInstance) AssignRoles(ctx context.Context, userID string, roles []en
 
 	return true, nil
 }
+
+// RevokeRoles revokes roles from a user
+func (db *PGInstance) RevokeRoles(ctx context.Context, userID string, roles []enums.UserRoleType) (bool, error) {
+	var (
```
Comment (KathurimaKimathi) [inline pkg/mycarehub/infrastructure/database/postgres/gorm/update.go:711]: why not just ``var user User``? I think its massacring readability

File: pkg/mycarehub/infrastructure/database/postgres/gorm/update.go [hunk 4]
```diff
@@ -704,3 +705,56 @@ func (db *PGInstance) AssignRoles(ctx context.Context, userID string, roles []en
...
+
+	err := tx.Model(&User{}).Where(&User{UserID: &userID}).First(&user).Error
+	if err != nil {
+		helpers.ReportErrorToSentry(err)
+		tx.Rollback()
+		return false, fmt.Errorf("failed to get user: %v", err)
+	}
+
+	for _, role := range roles {
+		var (
```
Comment (KathurimaKimathi) [inline pkg/mycarehub/infrastructure/database/postgres/gorm/update.go:734]: This too

**Review by Muchogoc — APPROVED** [review]

**Pull request merged**